src/components/menu/menu.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, MutableRefObject, PropsWithChildren } from 'react'
+import { useClickOutside } from '../../hooks/use-click-outside'
+import { MenuItemGroup } from './menu.atoms'
+import { MenuProvider, useMenu } from './provider'
+import { ElMenu, ElMenuItemContainer, ElMenuPopover } from './styles'
+
+const Trigger = ({ children }) => {
+  const { triggerProps, isOpen } = useMenu()
+
+  return children({ triggerProps, isOpen })
+}
+
+const Popover: FC<PropsWithChildren<{ containerRef: MutableRefObject<HTMLDivElement | null> }>> = ({
+  children,
+  containerRef,
+}) => {
+  const { isOpen, setIsOpen, popoverProps } = useMenu()
+
+  useClickOutside(containerRef as MutableRefObject<HTMLDivElement>, () => {
+    setIsOpen(false)
+  })
+
+  if (isOpen) {
+    const onClick = (e) => {
+      const button = e.target.closest('button')
+      if (button) setIsOpen(false)
+    }
+
+    return (
+      <ElMenuPopover {...popoverProps} onClick={onClick}>
+        {children}
+      </ElMenuPopover>
+    )
+  }
+
+  return null
+}
+
```

Comment:
**note:** I forgot to mention here... if we were to split up the MenuProvider into a simple context provider and a `useMenu` hook, our main `Menu` component here becomes the place where we tie the two together:

```javascript
function Menu({ children }) {
  const menu = useMenu()
  return (
    <Menu.Context.Provider value={menu}>
      {children}
    </Menu.Context.Provider>
  )
}
```